tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/SQS/ContextPropagation.cs

```diff
@@ -26,8 +26,21 @@ private static void Inject<TMessageRequest>(SpanContext context, IDictionary mes
             // Consolidate headers into one JSON object with <header_name>:<value>
             var sb = Util.StringBuilderCache.Acquire(Util.StringBuilderCache.MaxBuilderSize);
             sb.Append('{');
+
             SpanContextPropagator.Instance.Inject(context, sb, default(StringBuilderCarrierSetter));
-            dataStreamsManager?.InjectPathwayContext(context.PathwayContext, AwsSqsHeadersAdapters.GetInjectionAdapter(sb));
+
+            if (dataStreamsManager != null && dataStreamsManager.IsEnabled && context.PathwayContext != null)
+            {
+                var encodedBytes = PathwayContextEncoder.Encode(context.PathwayContext.Value);
+                var base64EncodedContext = Convert.ToBase64String(encodedBytes);
+                sb.AppendFormat("\"{0}\":\"{1}\",", DataStreamsPropagationHeaders.PropagationKeyBase64, base64EncodedContext);
+
+                if (Tracer.Instance.Settings.IsDataStreamsLegacyHeadersEnabled)
+                {
+                    sb.AppendFormat("\"{0}\":\"{1}\",", DataStreamsPropagationHeaders.PropagationKey, base64EncodedContext);
```

Comment:
Is this supposed to be `base64EncodedContext` or `encodedBytes`?  Right now there isn't anything different except for the key name
ex)

```suggestion
                    sb.AppendFormat("\"{0}\":\"{1}\",", DataStreamsPropagationHeaders.PropagationKey, encodedBytes);
```

Reply:
Great question! Yes, this is intended. I'd discussed this with Lucas briefly on slack but it is essentially because we want to support both the old and new key names for backwards compatibility.

Reply:
Great!
Could we add a comment then here saying that :)